### PR TITLE
Custom attributes support on analytics component

### PR DESCRIPTION
**Pull request by ethanclevenger91**
For things like `data-includes="credentials"` or remote config.

### CODE REVIEW

**Comment by ethanclevenger91** [comment]
@googlebot I signed it!

**Comment by westonruter** [comment]
Related: #1949 (adding `data-block-on-consent` attribute to `amp-analytics`)

File: includes/amp-helper-functions.php [hunk 1]
```diff
@@ -644,7 +644,7 @@ function amp_get_analytics( $analytics = [] ) {
 	foreach ( $analytics_entries as $entry_id => $entry ) {
 		$analytics[ $entry_id ] = [
 			'type'        => $entry['type'],
-			'attributes'  => [],
+			'attributes'  => array_merge( [], $entry['attributes'] ),
```
Comment (westonruter) [inline includes/amp-helper-functions.php]: Shouldn't this check to see if `isset( $entry['attributes'] )`?

Comment (westonruter) [reply includes/amp-helper-functions.php]: And no need to merge. I think it should rather be:

```php
'attributes'  => isset( $entry['attributes'] ) ? $entry['attributes'] : [],
```

**Comment by westonruter** [comment]
This should also include an update to the `\AMP_Analytics_Options_Test::test_amp_get_analytics()` unit test to ensure that `attributes` get added as expected. So something like this:

```diff
diff --git a/tests/php/test-amp-analytics-options.php b/tests/php/test-amp-analytics-options.php
index 36dfd3d3..97683902 100644
--- a/tests/php/test-amp-analytics-options.php
+++ b/tests/php/test-amp-analytics-options.php
@@ -241,11 +241,17 @@ class AMP_Analytics_Options_Test extends WP_UnitTestCase {
 			'amp_analytics_entries',
 			static function( $analytics ) use ( $key ) {
 				$analytics[ $key ]['type'] = 'test';
+
+				$analytics[ $key ]['attributes']['data-include']          = '_till_responded';
+				$analytics[ $key ]['attributes']['data-block-on-consent'] = 'credentials';
 				return $analytics;
 			}
 		);
 		$analytics = amp_get_analytics();
+
 		$this->assertEquals( 'test', $analytics[ $key ]['type'] );
+		$this->assertEquals( '_till_responded', $analytics[ $key ]['data-include'] );
+		$this->assertEquals( 'credentials', $analytics[ $key ]['data-block-on-consent'] );
 	}
 
 	/**
```

It wouldn't hurt to update `\AMP_Analytics_Options_Test::test_amp_print_analytics()` to also ensure that those attributes actually get added to the `amp-analytics` tag.

**Pull request closed without merging**